src/cpu/testers/dr_trace_player/trace_player.cc

```diff
@@ -167,24 +165,59 @@ DRTracePlayer::executeMemInst(DRTraceReader::TraceRef &mem_ref)
 bool
 DRTracePlayer::trySendMemRef(DRTraceReader::TraceRef &mem_ref)
 {
-    PacketPtr pkt = getPacket(mem_ref);
+    PacketPtr pkt, split_pkt;
+
+    std::tie(pkt, split_pkt) = getPacket(mem_ref);
+
+    if (!retrySplitPkt) {
+        // we should not be sending the first pkt again if
+        // we are retrying only on the second part of the
+        // previously stalled request
+        DPRINTF(DRTrace, "Trying to send %s\n", pkt->print());
+
+        if (!port.sendTimingReq(pkt)) {
+            DPRINTF(DRTrace, "Failed to send pkt\n");
+            if (stats.memStallStart == 0) {
+                stats.memStallStart = curTick();
+            }
+            numOutstandingMemReqs++;
+            stats.outstandingMemReqs.sample(numOutstandingMemReqs);
+
+            delete pkt;
+            if (split_pkt == nullptr) {
+            // if this is not a split pkt req,
+            // we can return back here
+            return true;
```

Comment:
I think we should delete the other packet and return when the first packet fails.

Reply:
Let me know if you disagree with this comment.

Reply:
Pushed a change to delete the split_pkt as well when the first pkt fails